src/components/ProtectedRoute/index.js

```diff
@@ -0,0 +1,13 @@
+import React, { lazy, createContext } from 'react';
+
+const Login = lazy(() => import('../../screens/Login'));
+
+export const AuthContext = createContext();
+
+const ProtectedRoute = ({ component: Component, ...rest }) => (
+  <AuthContext.Consumer>
```

Comment:
This is fine. I personally prefer another way of consuming context using hooks:
```
const ProtectedRoute = ({ component: Component, ...rest }) => {
  const { isAuth } = useContext(AuthContext);
  return isAuth ? <Component {...rest} /> : <Login />;
}
```

Mainly because you might want to process isAuth or whatever data you take from the context, and you'd have to do it inside the JSX. But that's a personal opinion rather than an improvement